internal/router/controllers/user.go

```diff
@@ -183,148 +179,74 @@ func (c *UserController) GetTableRow(rw http.ResponseWriter, r *http.Request) {
-	rows, ok := c.runReadRequest(r.Context(), stm, rw)
+	res, ok := c.runReadRequest(r.Context(), stm, rw)
 	if !ok {
 		return
 	}
 
-	rw.WriteHeader(http.StatusOK)
-	encoder := json.NewEncoder(rw)
-
-	modestr := r.URL.Query().Get("mode")
-	mode, ok := modeFromString(modestr)
-	if !ok {
-		mode = ColumnsMode
-	}
-	switch mode {
-	case ColumnsMode:
-		_ = encoder.Encode(rows)
-	case RowsMode:
-		_ = encoder.Encode(rows.Rows)
-	case JSONMode:
-		jsonrows := make([]map[string]interface{}, len(rows.Rows))
-		for i, row := range rows.Rows {
-			jsonrow := make(map[string]interface{}, len(row))
-			for j, col := range row {
-				jsonrow[rows.Columns[j].Name] = col
-			}
-			jsonrows[i] = jsonrow
-		}
-		_ = encoder.Encode(jsonrows)
-	case CSVMode:
-		rw.Header().Set("Content-type", "text/csv")
-		for i, col := range rows.Columns {
-			if i > 0 {
-				_, _ = rw.Write([]byte(","))
-			}
-			_, _ = rw.Write([]byte(col.Name))
-		}
-		_, _ = rw.Write([]byte("\n"))
-		for _, row := range rows.Rows {
-			for j, col := range row {
-				if j > 0 {
-					_, _ = rw.Write([]byte(","))
-				}
-
-				b, err := json.Marshal(col)
-				if err != nil {
-					rw.WriteHeader(http.StatusBadRequest)
-					log.Ctx(r.Context()).
-						Error().
-						Str("sql_request", stm).
-						Err(err)
-
-					_ = encoder.Encode(errors.ServiceError{Message: err.Error()})
-					return
-				}
-				_, _ = rw.Write(b)
-			}
-			_, _ = rw.Write([]byte("\n"))
-		}
-	case ListMode:
-		rw.Header().Set("Content-type", "application/jsonl+json")
-		for _, row := range rows.Rows {
-			for j, col := range row {
-				if j > 0 {
-					_, _ = rw.Write([]byte("|"))
-				}
-
-				b, err := json.Marshal(col)
-				if err != nil {
-					rw.WriteHeader(http.StatusBadRequest)
-					log.Ctx(r.Context()).
-						Error().
-						Str("sql_request", stm).
-						Err(err)
-
-					_ = encoder.Encode(errors.ServiceError{Message: err.Error()})
-					return
-				}
-				_, _ = rw.Write(b)
-			}
-			_, _ = rw.Write([]byte("\n"))
-		}
-	default:
+	opts, err := formatterOptions(r)
+	if err != nil {
 		rw.WriteHeader(http.StatusBadRequest)
-		err := fmt.Errorf("unrecognized mode: %s", modestr)
-		log.Ctx(r.Context()).
-			Error().
-			Str("mode", modestr).
-			Err(err)
+		msg := fmt.Sprintf("Error parsing formatting params: %v", err)
+		_ = json.NewEncoder(rw).Encode(errors.ServiceError{Message: msg})
+		log.Ctx(r.Context()).Error().Err(err).Msg(msg)
+		return
+	}
+	formatted, config, err := formatter.Format(res, opts...)
+	if err != nil {
+		rw.WriteHeader(http.StatusInternalServerError)
+		msg := fmt.Sprintf("Error formatting data: %v", err)
+		_ = json.NewEncoder(rw).Encode(errors.ServiceError{Message: msg})
+		log.Ctx(r.Context()).Error().Err(err).Msg(msg)
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
 
-		_ = encoder.Encode(errors.ServiceError{Message: err.Error()})
+	if config.Unwrap && len(res.Rows) > 1 {
+		rw.Header().Set("Content-Type", "application/jsonl+json")
 	}
+	_, _ = rw.Write(formatted)
```

Comment:
This is the handling for the general `/query` endpoint. Same logic as described above applies here.